src/js/api/posts/create.js

```diff
@@ -0,0 +1,30 @@
+import { dummyApiUrl } from "../constants.js";
+import { dummyApiCreatePost } from "../constants.js";
+import { authBaseFetchOpen } from "../apiBaseFetch.js";
+
+/**
+ * Creates a new listing
+ * @param {string} listData - Represents input information
+ */
+export async function create(listData) {
+  const createListingUrl = dummyApiUrl + dummyApiCreatePost;
+
+  console.log(createListingUrl)
+
+  const data = await authBaseFetchOpen(createListingUrl, {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(listData),
+  }
+  );
+
+  if (data.ok) {
+    // return await response.json();
+    const response = await data.json();
+    console.log(response);
+  } else {
+    console.log(data);
+    alert("Something went wrong, please try again");
```

Comment:
This is an alert for the user.
I would like an error response as well.
Instead of:
   } else {
    console.log(data);
    alert("Something went wrong, please try again");

I would like to see something like this:
 throw new Error("Something went wrong, please try again")

Reply:
Changed error handling to suggested.